compiler/optimizer/OSRDefAnalysis.cpp

```diff
@@ -1474,26 +1754,37 @@ void TR_OSRLiveRangeAnalysis::buildOSRLiveRangeInfo(TR::Node *node, TR_BitVector
    *_workBitVector -= *liveVars;
 
    TR_BitVector *deadSymRefs = NULL;
+   TR_BitVector deadSymRefsTemp(0, comp()->trMemory()->currentStackRegion());
+
    if (!_workBitVector->isEmpty())
       {
-      // Load the existing BitVector
-      deadSymRefs = osrMethodData->getLiveRangeInfo(byteCodeIndex);
-
       TR_BitVectorIterator bvi(*_workBitVector);
       while (bvi.hasMoreElements())
          {
          int32_t nextDeadVar = bvi.getNextElement();
          if (liveLocalIndexToSymRefNumberMap[nextDeadVar] >= 0)
-            {
-            if (!deadSymRefs)
-               {
-               deadSymRefs = new (trHeapMemory()) TR_BitVector(0, trMemory(), heapAlloc);
-               newlyAllocated = true;
-               }
-            deadSymRefs->set(liveLocalIndexToSymRefNumberMap[nextDeadVar]);
-            }
+            deadSymRefsTemp.set(liveLocalIndexToSymRefNumberMap[nextDeadVar]);
+         }
+
+      // Load the existing BitVector
+      deadSymRefs = osrMethodData->getLiveRangeInfo(byteCodeIndex);
+      if (!deadSymRefs)
+         {
+         deadSymRefs = new (trHeapMemory()) TR_BitVector(0, trMemory(), heapAlloc);
```

Comment:
The conditions for an allocation here have changed, which could result in more heap usage. Would it be possible to prevent an allocation if its know that the result will be an empty bit vector, by checking `deadSymRefsTemp `?

Reply:
Thanks for the suggestion. Changed the code to allocate deadSymRefs bit vector only when necessary.